tests/run_tests.py

```diff
@@ -98,16 +106,22 @@ def run_tests(test_suite):
         os.environ['MNE_BIDS_STUDY_CONFIG'] = config_path
         del config_name, config_path
 
-        # Fetch the data
+        # Fetch the data and remove potentially leftover derivatives e.g. from
+        # previously aborted tests.
         fetch(dataset)
+        clean_derivatives(os.environ['BIDS_ROOT'])
 
         # run the pipelines
         for pipeline in test_tuple[1::]:
             pipeline()
 
+        # Delete derivatives after testing is finished. Skip this step on
+        # circle, as we intend to retain the generated reports.
+        if os.environ['CI'] != 'true':
+            clean_derivatives(os.environ['BIDS_ROOT'])
```

Comment:
I think your changes to cricle.yml are enough

Reply:
I thought this here might be nice to have for running tests locally. Thoughts?